### PR TITLE
1999 introduce versioneer

### PR DESCRIPTION
**Pull request by jimmykarily**
Part of https://github.com/kairos-io/kairos/issues/1999

After this is merged, the `list-releases` command of kairos-agent will be re-implemented using the `versioneer` library.
The provider-kairos `ListVersions` will also be implemented using the same library, so the results will be consistent whether in a core or a standard Kairos image.

This library also implements all the functionality of the [naming.sh](https://github.com/kairos-io/kairos/blob/master/images/naming.sh) which means we can now replace that script everywhere. A PR to replace naming.sh should follow.

The library can be consumed either through the standalone cli (`bin/versioneer` in this repo) or through the kairos-agent where we will embed it under a `kairos-agent versioneer` command (https://github.com/kairos-io/kairos-agent/blob/89c3ab4f798138f0c4426cfd3fb319736ef1c443/main.go#L795-L800).

### CODE REVIEW

File: versioneer/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+_ -eer suffix :  (in nouns) a person concerned with a particular thing` (https://www.oxfordlearnersdictionaries.com)
+
+Versioneer is a library and a wrapping cli that is concerned with everything related to artifact names and versions in Kairos.
+
+TODO: usage
```
Comment (jimmykarily) [inline versioneer/README.md:5]: TODO

File: versioneer/cli.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+		{
+			Name:  "os-release-variables",
+			Usage: "generates a set of variables to be appended in the /etc/os-release file",
+			Flags: []cli.Flag{
+				flavorFlag, flavorReleaseFlag, variantFlag, modelFlag, archFlag, versionFlag,
+				softwareVersionFlag, registryAndOrgFlag, bugReportURLFlag, projectHomeURLFlag,
+				githubRepoFlag,
+			},
+			Action: func(cCtx *cli.Context) error {
+				a := Artifact{
```
Comment (Itxaka) [inline versioneer/cli.go:186]: this couold be extracted into a function to not duplicate it so much.

File: versioneer/versioneer.go [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+func (a *Artifact) TagList(registryAndOrg string) (TagList, error) {
+	if a.RegistryInspector == nil {
+		a.RegistryInspector = &DefaultRegistryInspector{}
+	}
+
+	return a.RegistryInspector.TagList(registryAndOrg, a)
+}
+
+func (a *Artifact) commonName() (string, error) {
+	if err := a.Validate(); err != nil {
```
Comment (Itxaka) [inline versioneer/versioneer.go:233]: we only validate here, should we do it in a lot more places?

**Review by Itxaka — APPROVED** [review]
just one question about the validation, rest looks good to me!

File: versioneer/versioneer.go (hunk 3 above)
Comment (jimmykarily) [reply versioneer/versioneer.go:233]: All the others are eventually calling this one. For some of them, I've added a test to make sure it is indeed validated. E.g. https://github.com/kairos-io/kairos-sdk/pull/59/files/d5a87ece6e62a02b547bac516ce1a0ba5b6e5183#diff-f8a8b10ab615219af502040ef4bfc01f8e25100aa2adf76de32a88a0277ccb6bR54

I probably didn't test that for all methods calling `commonName()` but all of them do.

**Pull request merged**